### ISSUE TITLE
girl have beard

### ISSUE THREAD

**Issue by tengshaofeng** [VinAIResearch/HyperInverter#4]
You really did a great job. Thanks for sharing. When test some image, I found when person become old,  person will wear glass, some girl have beard. How to solve that?
![screenshot-20220616-104340](https://user-images.githubusercontent.com/9246739/178405716-8b53e4f9-7a33-4c65-80ec-63a365742d78.png)
![8b927126db85d175fb8cf6b69724579d](https://user-images.githubusercontent.com/9246739/178405803-1f0fb4e6-6e66-4560-92d6-b4172e1503bd.jpg)

**Comment by di-mi-ta** [comment]
Hi, 
Thanks for your interest in our work!
To answer the question, for age editing direction, we leverage from [PTI repo](https://github.com/danielroich/PTI/tree/main/editings/interfacegan_directions). Specifically, the age editing direction learned by InterfaceGAN on StyleGAN2 trained on the FFHQ dataset often adds glass when editing face images toward old. About the entangle phenomenon for adding beard when editing face image more aged, this issue also appears when we only use W-Encoder without our phase II refinement. We assume that the current age edit direction does not work well with the latent code inveted by phase I pSp/e4e W encoder. To mitigate this problem, learning better age editing direction may help.

**Comment by tengshaofeng** [comment]
@di-mi-ta

Ok, Thanks for your advice. By the way, how can I learn a better age editing direction. I have two ideas below.
1.  Generate z randomly, and map to w, then synthesis the image given pretrained stylegan2 model from [stylegan2-ada-pytorch](https://github.com/NVlabs/stylegan2-ada-pytorch), then label the image with smile/not smile , then learn the editing direction.

2. Given many real-world images, then encode them to w given pSp encoder model. Then do as same as the first method.

Is it right? which is better?
